Add documentation for slab benchmark multiphysics problem. (#99)
idaholab/salamander: doc/content/verification_validation_examples/slab_benchmark/slab_benchmark.md

# 1D Slab Benchmark: Neutron Transport Coupled With Thermal Conduction and Material Expansion

This verification problem presents capabilities to couple neutron transport with thermal conduction and material expansion. The analytic benchmark slab problem [!citep](griesheimer2022analytical) describes a slab where S$_2$ neutron transport is coupled with thermal conduction, convective boundary conditions, Doppler-broadened cross sections, and nonlinear thermal expansion effects along the heated slab. The benchmark provides the analytic solutions for steady-state temperature, neutron flux, heating, multiplication factor, and the heated slab length.

> While the restriction to S$_2$ neutron transport would be considered overly simplistic for most radiation transport  simulations, the objective here is to verify thermomechanical-OpenMC coupling using analytic solutions for three coupled solution fields (Displacements, temperature, and flux). Simplifications of each individual physics included in this model are not imposed by inherent limitations of the moving-mesh capabilities and only aim to allow the study on the interactions between multiphysics with an analytic solution.

Reviewer: ```suggestion
While the restriction to S$_2$ neutron transport would be considered overly simplistic for most radiation transport  simulations, the objective here is to verify thermomechanical-OpenMC coupling using analytic solutions for three coupled solution fields (displacements, temperature, and flux). Simplifications of each individual physics included in this model are not imposed by inherent limitations of the moving-mesh capabilities and only aim to allow the study on the interactions between multiphysics with an analytic solution.
```

\end{equation}

where $\kappa(T)$ is the thermal conductivity. The thermal conductivity is described by a linear function of temperature as

\begin{equation}
  \kappa(T) = \kappa_0 T(x).
  \label{eqn:conductivity}
\end{equation}

The differential length at every point along the slab expands upon heating, where temperature is also varying, in proportion to the local thermal strain $\epsilon_x$, such as

\begin{equation}
  \ell(x) = \ell_0 (\epsilon_x +1),
  \label{eqn:ell_eqn}
\end{equation}

where $\ell(x)$ and $\ell_0$ are differential length elements in the heated and unheated slab, respectively. The thermal strain at any location $x$ is given by

\begin{equation}
  \epsilon_x(T)=\int_{T_0}^T \alpha(T) dT,
  \label{eqn:epsilon}
\end{equation}

where the linear expansion coefficient $\alpha(T)$ is given by an inverse square root dependence on temperature,

\begin{equation}
  \alpha(T)\equiv\frac{1}{2\sqrt{T_0T}}.
  \label{eqn:alpha}
\end{equation}

We use total mass preservation to get material density change from the initial to the final state, starting from:

\begin{equation}

>   \rho(x)\ell(x) = \rho_0\ell_0. \label{eqn:mass_conservation}

Reviewer: ```suggestion
  \rho(x)\ell(x) = \rho_0\ell_0, \label{eqn:mass_conservation}
```

\end{equation}

> To get the relation

Reviewer: ```suggestion
to get the relation
```

\begin{equation}
  \label{eqn:rho_t}

>   \rho(x) = \rho_0\sqrt{\frac{T_0}{T(x)}}

Reviewer: ```suggestion
  \rho(x) = \rho_0\sqrt{\frac{T_0}{T(x)}}.
```

\end{equation}

## Analytic solutions

> the reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) for full details on the derivation of these solution. Here we present the final formulas that we use later for comparison with our models.

Reviewer: ```suggestion
The reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) for full details on the derivation of these solutions. Here we present the final formulas that we use later for comparison with our models.
```

> The reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) for full details on the derivation of these solutions. Here we present the final formulas that we use later for comparison with our models.

Reviewer: ```suggestion
The reader is referred to the original benchmark publication [!citep](griesheimer2022analytical) and the updates made in [!citep](eltawila2025) for full details on the derivation of these solutions. Here we present the final formulas that we use later for comparison with our models.
```

\end{equation}

The final length of the heated slab $L$ is then given by

\begin{equation}
  L = \sqrt[3]{\Bigg(\frac{L_0}{_2F_1(1/4,1/2,3/2,1/\lambda)}\Bigg)^4\Bigg(\frac{\frac{q^2\phi^2_0}{P(\lambda-1)}-\frac{P}{4}}{\kappa_0T_0^2(1-1/\lambda)}\Bigg)}
  \label{eqn:length},
\end{equation}

where $_2F_1$ is the Gauss hypergeometric function.

To ensure a physically meaningful solution (i.e., the heat flux leaving the slab is consistent with the $T(\pm L/2)-T_0$), there is an additional constraint on the manufactured temperature distribution

\begin{equation}
  T\left(\pm\frac{L}{2}\right)\geq T_0.
  \label{eqn:constraint}
\end{equation}

Substituting [eqn:flux_solution] and [eqn:temperature] into [eqn:constraint] gives a minimum condition for power in terms of the other slab parameters, as

\begin{equation}
  P \geq \frac{4\kappa_0T_0^2}{L(\lambda-1)}.
  \label{eqn:power_condition}
\end{equation}

The example set of model parameter values shown in [tab:params] are the suggested benchmark parameters [!citep](griesheimer2022analytical), resulting in $k_\text{eff} = 0.29557$ and a heated slab length of $L = 106.47$ `cm` with $P = 1.0\times10^{22}$ `eV/s`.

> !table id=tab:params caption=Example set of model parameters with two different power scenarios [!citep](griesheimer2022analytical).

Reviewer: You describe two power scenarios here, but you only list one in the table. Make sure to be consistent.

> | **Parameter                  | **Value                   | **Unit                              |

Reviewer: Why do you use the `**` symbols everywhere? What do they do? 
I also don't see the need for the ` signs around the units.

| :- | :- | :- |
| **$\rho_0$**                   | 1.2                         | `[g.cm`$^{-3}$`]`                          |
| **$L_0$**                      | 100                         | `[cm]`                                  |
| **$A$**                        | 180                         | `[g/mol`$^{-1}$`]`                        |
| **$T_0$**                      | 293                         | `[K]`                                   |

> | **$q$**                        | 1.0E8                       | `[eV]`                                  |

Reviewer: ```suggestion
| **$q$**                        | 1.0 $\times$ 10$^{8}$           | `[eV]`                                  |
```

> | **$\kappa_0$**                 | 1.25E19.                    | `[eV/s/cm/K`$^{-2}$`]`      |

Reviewer: ```suggestion
| **$\kappa_0$**                 | 1.25 $\times$ 10$^{19}$.                    | `[eV/s/cm/K`$^{-2}$`]`      |
```

> | **$\phi_0$**                   | 2.5E14                      | `[#/cm`$^2$`/s]`                  |

Reviewer: ```suggestion
| **$\phi_0$**                   | 2.5 $\times$ 10$^{24}$        | `[#/cm`$^2$`/s]`                  |
```

| **$\nu\Sigma_f/\Sigma_t$**     | 1.5                         | `[-]`                                   |
| **$\Sigma_s/\Sigma_t$**        | 0.45                        | `[-]`                                   |

> | **$P$**                        | 1.0E22                      | `[eV/s]`                         |

Reviewer: ```suggestion
| **$P$**                        | 1.0 $\times$ 10$^{22}$        | `[eV/s]`                         |
```

# Problem model

!alert construction title=To run this model you need to build your application with the S$_2$ OpenMC patch
This model is using S$_2$ neutron transport to compare with the 1D Analytic benchmark. To replicate
the results, you have build your application with [this](https://github.com/meltawila/openmc_S2transport) patch of OpenMC.

> Cardinal is used to couple OpenMC with MOOSE via Picard iteration. OpenMC tallies neutron flux, heating, and the multiplication factor. A conceptual depiction of the meshes used as well as the data transfers that occur on each Picard iteration is shown in [fig:schematic]. Three different meshes/geometries are involved. The thermomechanics mesh is a simple 1D mesh of $N$ EDGE2 elements; the thermomechanical physics do not require higher dimensional (e.g. 2D or 3D) meshes for this problem. The OpenMC geometry is represented using DAGMC, where cells are bounded by a triangulated surface (TRI3 elements). To facilitate data transfers between OpenMC and MOOSE thermomechanics, there is also an intermediate volumetric data transfer mesh composed of TET4 elements; no physics solves happen on this intermediate mesh. [fig:schematic] highlights one element (for MOOSE and the Cardinal intermediate data transfer mesh) and one OpenMC cell each in blue, with adjacent portions of the geometry shown in gray for context.

Reviewer: ```suggestion
Cardinal is used to couple OpenMC with MOOSE via Picard iteration. OpenMC tallies neutron flux, heating, and the multiplication factor. A conceptual depiction of the meshes used as well as the data transfers that occur on each Picard iteration is shown in [fig:schematic]. 
Three different meshes/geometries are involved. The thermomechanics mesh is a simple 1D mesh of $N$ EDGE2 elements; the thermomechanical physics do not require higher dimensional (e.g., 2D or 3D) meshes for this problem. The OpenMC geometry is represented using DAGMC, where cells are bounded by a triangulated surface (TRI3 elements). 
To facilitate data transfers between OpenMC and MOOSE thermomechanics, there is also an intermediate volumetric data transfer mesh composed of TET4 elements; no physics solves happen on this intermediate mesh. 
[fig:schematic] highlights one element (for MOOSE and the Cardinal intermediate data transfer mesh) and one OpenMC cell each in blue, with adjacent portions of the geometry shown in gray for context.
```

> Cardinal is used to couple OpenMC with MOOSE via Picard iteration. OpenMC tallies neutron flux, heating, and the multiplication factor. A conceptual depiction of the meshes used as well as the data transfers that occur on each Picard iteration is shown in [fig:schematic]. Three different meshes/geometries are involved. The thermomechanics mesh is a simple 1D mesh of $N$ EDGE2 elements; the thermomechanical physics do not require higher dimensional (e.g. 2D or 3D) meshes for this problem. The OpenMC geometry is represented using DAGMC, where cells are bounded by a triangulated surface (TRI3 elements). To facilitate data transfers between OpenMC and MOOSE thermomechanics, there is also an intermediate volumetric data transfer mesh composed of TET4 elements; no physics solves happen on this intermediate mesh. [fig:schematic] highlights one element (for MOOSE and the Cardinal intermediate data transfer mesh) and one OpenMC cell each in blue, with adjacent portions of the geometry shown in gray for context.

Reviewer: ```suggestion
Cardinal is used to couple OpenMC with MOOSE via Picard iteration. OpenMC tallies neutron flux, heating, and the multiplication factor. A conceptual depiction of the meshes used as well as the data transfers that occur on each Picard iteration is shown in [fig:schematic]. Three different meshes/geometries are involved. The thermomechanics mesh is a simple 1D mesh of $N$ EDGE2 elements; the thermomechanical physics do not require higher dimensional (e.g., 2D or 3D) meshes for this problem. The OpenMC geometry is represented using DAGMC, where cells are bounded by a triangulated surface (TRI3 elements). To facilitate data transfers between OpenMC and MOOSE thermomechanics, there is also an intermediate volumetric data transfer mesh composed of TET4 elements; no physics solves happen on this intermediate mesh. [fig:schematic] highlights one element (for MOOSE and the Cardinal intermediate data transfer mesh) and one OpenMC cell each in blue, with adjacent portions of the geometry shown in gray for context.
```

The main sources of error in this coupled model are: (i) the statistical error from use of finite particles in OpenMC; (ii) the user-input cross-section data, which is defined with 1 K $\Delta$T spacings used in all cases while taking the nearest temperature point cross section; and (iii) discretization error from a finite spatial mesh in both OpenMC (temperature feedback resolution) and MOOSE thermomechanics. Additional sources of error are present, but anticipated to be negligible due to the use of fine settings for these simulation parameters: (i) finite Picard iterations, and (ii) nonzero nonlinear tolerances in the MOOSE thermomechanics model.

## Results

> We see the heating result in [fig:heatingsol] compared against the analytic solution, as well as flux and temperature solutions in [fig:fluxsol] and [fig:tempsol]. Figures are only included for P=1.0E22 `[eV/s]` and on a fine mesh where the number of OpenMC and mesh elements is $N = 100$, since all results have a similar shape. Both the heating and flux tally everywhere agrees with the analytic solution within 3σ.

Reviewer: ```suggestion
We see the heating result in [fig:heatingsol] compared against the analytic solution, as well as flux and temperature solutions in [fig:fluxsol] and [fig:tempsol]. Figures are only included for $P=1.0 \times 10^{22}$ eV/s and on a fine mesh where the number of OpenMC and mesh elements is $N = 100$, since all results have a similar shape. Both the heating and flux tally everywhere agrees with the analytic solution within 3$\sigma$.
```

!media figures/slab_benchmark_heatingsol.png
  id=fig:heatingsol
  caption=Heating result from OpenMC for N = 100 mesh elements compared against the analytic solution.
  style=display:block;margin-left:auto;margin-right:auto;width:40%;

!media figures/slab_benchmark_fluxsol.png
  id=fig:fluxsol
  caption=Flux result from OpenMC for N = 100 mesh elements compared against the analytic solution.
  style=display:block;margin-left:auto;margin-right:auto;width:40%;

!media figures/slab_benchmark_tempsol.png
  id=fig:tempsol
  caption=Temperature result from MOOSE for N = 100 mesh elements compared against the analytic solution.
  style=display:block;margin-left:auto;margin-right:auto;width:40%;

> Table [tab:results] shows results convergence with mesh refinement.

Reviewer: ```suggestion
[tab:results] shows results convergence with mesh refinement.
```
You shouldn't need `Table` here.

!table id=tab:results caption=Results and corresponding errors with different mesh sizes [!citep](eltawila2025).

> | **Resolution        | **Heated length `[cm]` | **Error `[μm]` | **$k_{\text{eff}}$** | **Error `[pcm]` |

Reviewer: Same comment about ** and `

| :- | :- | :- | :- | :- |
| **Analytic solution | 106.47                 | --             | 0.29557              | --              |

> | **5                 | 106.3298               | -1402          | 0.29608 ± 0.00001    | 51 ± 1          |

Reviewer: Once the documentation builds, I'll need to check that the plus/minus signs work in markdown.

| **10                | 106.4382               | -318           | 0.29619 ± 0.00001    | 62 ± 1          |
| **20                | 106.4671               | -29            | 0.29557 ± 0.00001    | 0 ± 1           |
| **50                | 106.4730               | 30             | 0.29536 ± 0.00001    | -21 ± 1         |
| **100               | 106.4745               | 45             | 0.29549 ± 0.00001    | -8 ± 1          |
| **200               | 106.4748               | 48             | 0.29556 ± 0.00001    | -1 ± 1          |

> In [fig:fluxconv] and [fig:tempconv] convergence of flux and temperature relative L$^2$ norms, defined as

Reviewer: ```suggestion
In [fig:fluxconv] and [fig:tempconv], convergence of flux and temperature relative L$^2$ norms, defined as
```

  \label{eqn:power_condition}
\end{equation}

> The example set of model parameter values shown in [tab:params] are the suggested benchmark parameters [!citep](griesheimer2022analytical), resulting in $k_\text{eff} = 0.29557$ and a heated slab length of $L = 106.47$ `cm` with $P = 1.0\times10^{22}$ `eV/s`.

Reviewer: General comment: I do not think you need the "`" signs around units. We usually do not use them in papers and in the rest of the documentation. For consistency, I recommend removing them.
